### PR TITLE
Minion network

### PR DESCRIPTION
**Pull request by mmussomele**
Change ports to have the same name as their IP and rework network.go:runMaster to use a hashjoin

### CODE REVIEW

File: minion/network/network.go [hunk 1]
```diff
@@ -63,77 +79,78 @@ func runMaster(conn db.Conn) {
 		})
 
 		connections = view.SelectFromConnection(nil)
+
+		for _, l := range labels {
```
Comment (ejj) [inline minion/network/network.go]: Why do this within the transaction?

File: minion/network/network.go [hunk 2]
```diff
@@ -63,77 +79,78 @@ func runMaster(conn db.Conn) {
...
+
 		return nil
 	})
 
 	if !leader {
 		return
 	}
 
-	ovsdb, err := ovsdb.Open()
+	openvsdb, err := ovsdb.Open()
```
Comment (ejj) [inline minion/network/network.go]: This change isn't appropriate for this commit.  Could do it in a later one if you'd like, though I think it was fine as it was.

File: minion/network/network.go [hunk 3]
```diff
@@ -63,77 +79,78 @@ func runMaster(conn db.Conn) {
...
-		garbageMap[lport.Name] = struct{}{}
+	portKey := func(val interface{}) interface{} {
+		return val.(ovsdb.LPort).Name
 	}
 
-	for _, dbl := range labels {
-		if !dbl.MultiHost {
-			continue
+	dbKey := func(val interface{}) interface{} {
+		switch val := val.(type) {
```
Comment (ejj) [inline minion/network/network.go]: The join should also include the bridge for future proofing.

**Comment by ejj** [comment]
Merged the first patch, some high level comments on the second patch (also some in slack).  Someone else should review as well.

File: minion/network/network.go [hunk 4]
```diff
@@ -64,79 +79,78 @@ func runMaster(conn db.Conn) {
...
-			}).Warn("Failed to create port.")
+	for _, dbp := range dbps {
+		switch dbp := dbp.(type) {
+		case db.Label:
+			if !dbp.MultiHost {
+				continue
+			}
+			makePort(lSwitch, dbp.IP, labelMac, dbp.IP)
+		case db.Container:
+			makePort("quilt", dbp.IP, dbp.Mac, dbp.IP)
```
Comment (sirspinach) [inline minion/network/network.go]: I think this should be `lSwitch` again, rather than `"quilt"`.

File: minion/network/network.go [hunk 5]
```diff
@@ -64,79 +79,78 @@ func runMaster(conn db.Conn) {
...
+		default:
+			panic("Not Reached")
 		}
+	}
 
-		if _, ok := garbageMap[dbl.Label]; ok {
-			delete(garbageMap, dbl.Label)
-			continue
-		}
+	_, ovsps, dbps := join.HashJoin(ovsdb.LPortSlice(lports), dbData, portKey, dbKey)
```
Comment (sirspinach) [inline minion/network/network.go]: I think ovsPorts is easier to read than ovsps. Your call here.

**Pull request merged**